staging/src/k8s.io/apimachinery/pkg/util/cache/cache.go

```diff
@@ -32,9 +32,10 @@ func NewCache(maxSize int) Cache {
 	}
 	cache := make(Cache, shardsCount)
 	for i := 0; i < shardsCount; i++ {
+                size := maxSize / shardsCount
 		cache[i] = &cacheShard{
-			items:   make(map[uint64]interface{}),
-			maxSize: maxSize / shardsCount,
+			items:   make(map[uint64]interface{}, size),
+			maxSize: size,
```

Comment:
if the max size is large, I think it's actually better to let it grow over time rather than preallocate this

Reply:
Presizing is common practice in k8s.
Is there guideline on the threshold beyond which presizing should not be done ?

Reply:
Presizing is good when it immediately precedes filling the allocated list/map with items of a known length. 

In this case, it would allocate memory that might never be used

Reply:
Looking at the sharding, it seems for heavily populated shard(s), there would be more evictions compared to less populated shard(s).

I wonder if there is better way of caching so that efficiency is higher.